src/diagnostics/logPointsManager.ts

```diff
@@ -0,0 +1,298 @@
+} from './logPointsClient';
+
+import WebSiteManagementClient = require('azure-arm-website');
+
+let shouldUseMockKuduCall = true;
+
+let logPointsDebuggerClient: LogPointsDebuggerClient;
+
+if (shouldUseMockKuduCall) {
+    logPointsDebuggerClient = new MockLogpointsDebuggerClient();
+} else {
+    logPointsDebuggerClient = new KuduLogPointsDebuggerClient();
+}
+
+export class LogPointsSessionAttach extends WizardBase {
+    private _cachedPublishCredential: WebSiteModels.User
+
+    public readonly hasSlot: boolean;
+    public selectedDeploymentSlot: WebSiteModels.Site;
+    public selectedInstance: WebSiteModels.SiteInstance;
+    public sessionId: string;
+    public processId: string;
+    public debuggerId: string;
+
+    constructor(output: vscode.OutputChannel,
+        readonly azureAccount: AzureAccountWrapper,
+        readonly site: WebSiteModels.Site,
+        readonly subscription?: SubscriptionModels.Subscription
+    ) {
+        super(output);
+        if (util.isSiteDeploymentSlot(this.site)) {
+            this.selectedDeploymentSlot = this.site;
+        } else {
+            this.steps.push(new PromptSlotSelection(this, this.site));
+        }
+
+        this.steps.push(new GetUnoccupiedInstance(this));
+        this.steps.push(new PickProcessStep(this));
+        this.steps.push(new SessionAttachStep(this));
+        this.steps.push(new StartDebugAdapterStep(this));
+    }
+
+    public get lastUsedPublishCredential(): WebSiteModels.User {
+        return this._cachedPublishCredential;
+    }
+
+    public get webManagementClient() {
+        return new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+    }
+
+    public async fetchPublishCrentential(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        const siteClient = new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+        const user = await util.getWebAppPublishCredential(siteClient, site);
+        this._cachedPublishCredential = user;
+        return user;
+    }
+
+    public async getCachedCredentialOrRefetch(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        if (this.lastUsedPublishCredential) {
+            return Promise.resolve(this.lastUsedPublishCredential);
+        }
+
+        return this.fetchPublishCrentential(site);
+    }
+
+    protected beforeExecute() { }
+
+    protected onExecuteError(error: Error) {
+        if (error instanceof UserCancelledError) {
+            return;
+        }
+        this.writeline(`Starting Log Points Session failed - ${error.message}`);
+        this.writeline('');
+    }
+}
+
+class PromptSlotSelection extends WizardStep {
+    constructor(private _wizard: LogPointsSessionAttach, readonly site: WebSiteModels.Site) {
+        super(_wizard, 'Choose a deployment slot.');
+    }
+
+    public async execute(): Promise<void> {

```

Comment:
Should most of this method be in the "prompt" method instead of execute?

Same for other wizard steps.

Reply:
Changed. I keep the last two steps using `execute()`, since they do not require user interaction.